src/util/triplea/MapXMLCreator/MapXMLHelper.java

```diff
@@ -0,0 +1,1008 @@
+			game.appendChild(resourceList);
+			for (final String resourceName : s_resourceList) {
+				final Element resource = doc.createElement("resource");
+				resource.setAttribute("name", resourceName);
+				resourceList.appendChild(resource);
+			}
+		}
+
+		if (!s_playerName.isEmpty()) {
+			Element playerList = doc.createElement("playerList");
+			game.appendChild(playerList);
+			playerList.appendChild(doc.createComment(" In Turn Order "));
+			for (final String playerName : s_playerName) {
+				final Element player = doc.createElement("player");
+				player.setAttribute("name", playerName);
+				player.setAttribute("optional", "false");
+				playerList.appendChild(player);
+			}
+
+			final HashMap<String, ArrayList<String>> alliances = new HashMap<String, ArrayList<String>>();
+			for (final Entry<String, String> allianceEntry : s_playerAlliance.entrySet()) {
+				final String allianceName = allianceEntry.getValue();
+				ArrayList<String> players = alliances.get(allianceName);
+				if (players == null) {
+					players = new ArrayList<String>();
+					alliances.put(allianceName, players);
+				}
+				players.add(allianceEntry.getKey());
+			}
+			for (final Entry<String, ArrayList<String>> allianceEntry : alliances.entrySet()) {
+				final String allianceName = allianceEntry.getKey();
+				playerList.appendChild(doc.createComment(" " + allianceName + " Alliance "));
+				for (final String playerName : allianceEntry.getValue()) {
+					final Element player = doc.createElement("alliance");
+					player.setAttribute("player", playerName);
+					player.setAttribute("alliance", allianceName);
+					playerList.appendChild(player);
+				}
+			}
+		}
+
+		if (!s_unitDefinitions.isEmpty()) {
+			map.appendChild(doc.createComment(" Unit Definitions "));
+			final Element unitList = doc.createElement("unitList");
+			game.appendChild(unitList);
+
+			final Element production = doc.createElement("production");
+			game.appendChild(production);
+			final String firstResourceName = s_resourceList.get(0);
+			for (final Entry<String, ArrayList<Integer>> unitDefinition : s_unitDefinitions.entrySet()) {
+				final String unitName = unitDefinition.getKey();
+
+				final Element unit = doc.createElement("unit");
+				unit.setAttribute("name", unitName);
+				unitList.appendChild(unit);
+
+				final ArrayList<Integer> definition = unitDefinition.getValue();
+				final Element productionRule = doc.createElement("productionRule");
+				productionRule.setAttribute("name", "buy" + unitName.substring(0, 1).toUpperCase() + unitName.substring(1));
+				production.appendChild(productionRule);
+				final Element prCost = doc.createElement("cost");
+				prCost.setAttribute("resource", firstResourceName);
+				prCost.setAttribute("quantity", definition.get(0).toString());
+				productionRule.appendChild(prCost);
+				final Element prResult = doc.createElement("result");
+				prResult.setAttribute("resourceOrUnit", unitName);

```

Comment:
This method is is many pages, could very much breaking up, likely into multiple classes (but there is a lot of space for elimination of redundant code)